### ISSUE TITLE
useWebSocket - possibility to update connection settings

### ISSUE THREAD

**Issue by Avaray** [vueuse/vueuse#1722]
### Clear and concise description of the problem

Websocket client works great, but I'm unable to change connection settings.
I wish to be able to pass new settings into open() function, something like:
````js
open(new_address, {new_options})
````

**or**
It would be great to have a **new** function like
````js
update(new_address, {new_options})
````
...and it will update current connection settings and reconnect. Something like that.

### Suggested solution

Provide possibility to update current connection settings.

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by Zehir** [comment]
Or maybe just a url that is a `Ref<string>` instead of a `string`.
Is there an usage of updating the options ? Most of them are at initialisation

**Comment by Avaray** [comment]
> Or maybe just a url that is a `Ref<string>` instead of a `string`. Is there an usage of updating the options ? Most of them are at initialisation

I think your idea is also good. 
Both are good in my opinion.

**Comment by lumdzeehol** [comment]
> Or maybe just a url that is a `Ref<string>` instead of a `string`. Is there an usage of updating the options ? Most of them are at initialisation

After websocket url changed, incidental effects might change either. It's likely to happen that heartbeat message being different in the new websocket service. Consequently I think options should be compatible with  `Ref / Reative` either

**Comment by Avaray** [comment]
Yay! We have it!
[https://github.com/vueuse/vueuse/pull/2367](https://github.com/vueuse/vueuse/pull/2367)

**Comment by ilyaliao** [comment]
Closed via #2367
